popsycle/synthetic.py

```diff
@@ -3512,3 +3527,402 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+    interpreted_rms_velocities = np.interp(rand_cdf, cdf, v_vals)
+
+    #Sampling random latitude and longitude values for velocity to complete the spherical velocities
+    sin_lat_vel = np.random.uniform(-1, 1, len(data_in_field))
+    lat_vel = np.arcsin(sin_lat_vel)
+    long_vel = np.random.uniform(0, 2*np.pi, len(data_in_field))
+
+    #Transforming velocities to cartesian to get vx, vy, vz values
+    cart_vel = astropy.coordinates.spherical_to_cartesian(interpreted_rms_velocities, lat_vel, long_vel)
+    vx= cart_vel[0]
+    vy= cart_vel[1]
+    vz= cart_vel[2]
+
+    #Getting rest of data for PBHs for the combined .h5 file
+    mass = np.full(len(data_in_field), pbh_mass)
+    zams_mass = mass
+    age = np.full(len(data_in_field), 14*(10**9))
+    pop_id = np.full(len(data_in_field), 10)
+    rem_id = np.full(len(data_in_field), 104)
+
+    b_rad = np.radians(b_in_field)
+    l_rad = np.radians(l_in_field)
+
+    cart_helio = astropy.coordinates.spherical_to_cartesian(r_in_field, b_rad, l_rad)
+
+    px = cart_helio[0]
+    py = cart_helio[1]
+    pz = cart_helio[2]
+
+    vr, mu_b, mu_lcosb = calc_sph_motion(vx, vy, vz, r_in_field, b_in_field, l_in_field)
+
+    exbv = np.full(len(data_in_field), 0)
+    ubv_k = np.full(len(data_in_field), np.nan)
+    ubv_j = np.full(len(data_in_field), np.nan)
+    obj_id = np.arange(1, (len(data_in_field)+1))
+    ubv_i = np.full(len(data_in_field), np.nan)
+    ubv_u = np.full(len(data_in_field), np.nan)
+    ubv_r = np.full(len(data_in_field), np.nan)
+    ubv_b = np.full(len(data_in_field), np.nan)
+    ubv_h = np.full(len(data_in_field), np.nan)
+    ubv_v = np.full(len(data_in_field), np.nan)
+
+    pbh_data = pd.DataFrame({'zams_mass':zams_mass, 'rem_id':rem_id, 'mass':mass, 'px':px, 'py':py, 'pz':pz, 'vx':vx, 'vy':vy, 'vz':vz, 'rad':r_in_field, 'glat':b_in_field, 'glon':l_in_field, 'vr':vr, 'mu_b':mu_b, 'mu_lcosb':mu_lcosb, 'age':age, 'popid':pop_id, 'ubv_k':ubv_k, 'ubv_i':ubv_i, 'exbv':exbv, 'obj_id':obj_id, 'ubv_j':ubv_j, 'ubv_u':ubv_u, 'ubv_r':ubv_r, 'ubv_b':ubv_b, 'ubv_h':ubv_h, 'ubv_v':ubv_v})
+
+    #Getting information out of .h5 file before PBHs
+    no_pbh_hdf5_file = h5py.File(hdf5_file, 'r')
+
+    for key in key_list:
+        keys = pd.DataFrame(np.array(no_pbh_hdf5_file['{}'.format(key)]))
+        vars()[key+'_max_l'] = np.amax(keys.T[11])
+        vars()[key+'_min_l'] = np.amin(keys.T[11])
+        vars()[key+'_max_b'] = np.amax(keys.T[10])
+        vars()[key+'_min_b'] = np.amin(keys.T[10])
+        vars()[key+'_mask'] = (pbh_data['glon'] >= vars()[key+'_min_l']) & (pbh_data['glon'] <= vars()[key+'_max_l']) & (pbh_data['glat'] >= vars()[key+'_min_b']) & (pbh_data['glat'] <= vars()[key+'_max_b'])
+        vars()['pbh_'+key] = (pbh_data[vars()[key+'_mask']]).T
+        vars()['pbh_'+key].reset_index(drop=True, inplace=True)
+        vars()['full_'+key]=pd.concat([keys, vars()['pbh_'+key]], axis=1)
+    no_pbh_hdf5_file.close()
+
```

Comment:
What we want to do is take each key in the hdf5 file, and append rows for all of the PBHs that this function generates. What this section of code does is add new keys with PBH information. Look at the documentation for perform_pop_syn and make sure that you have the right information in the right columns and replace the keys.